### DISCUSSION TITLE
Questions about eCommerce

### DISCUSSION THREAD

**Discussion by fusionconsulting** [area17/twill#1700] [General]
I'm building a website that will need a CMS so Twill looks perfect but I also need to add eCommerce functionality to it. I can do this the manual way and build a shopping cart system if necessary but wondering if there's any recommendations on adding eCommerce to Twill.

I would need to add products to categories as well. Is there a way to link modules? If I create a category module, can I then link this to a product module?

**Comment by ifox** [comment]
Hi @fusionconsulting, glad you found Twill!

I'll answer your second question first. Twill is all about linking/relationships between modules. From the most basic select field to more complex fields like browsers and repeaters, Twill is equipped to manage all Laravel Eloquent relationship types, with a nice UI for editors.

Regarding e-commerce, you can certainly use Twill to manage your products, categories, and even orders, but it is not in the scope of Twill to help with the frontend aspect. By frontend I mean frontend application, not just html/css/js but also your "frontend" routes and controllers in your Laravel app. There are many Laravel packages for things like cart management, though I can't vouch for a specific one, as we've built that ourself in our implementations.

**Reply by fusionconsulting** [reply]
Thanks for the reply. I'm digging into the documentation and seeing how it all fits together. Looks like it should be a great platform to work with. 

One question I have so far regarding nesting. I have run "composer require kalnoy/nestedset" to install the nesting module, however when I run "php artisan twill:module category --hasSlug --hasMedias --hasPosition --hasNesting" I get the following:

" The "--hasNesting" option does not exist."

Is there something further I have to do to enable nesting? I'm working through the documentation but it only seems to require that I install the nesting module.

**Reply by ifox** [reply]
Try with `twill:make:module`. `twill:module` is deprecated, we'll make sure to update the docs!
